rest/service/lastVisitedService.go

```diff
@@ -3,68 +3,36 @@ package service
 import (
 	"github.com/RedHatInsights/chrome-service-backend/rest/database"
 	"github.com/RedHatInsights/chrome-service-backend/rest/models"
-	"github.com/RedHatInsights/chrome-service-backend/rest/util"
-	"gorm.io/gorm"
-	"gorm.io/gorm/clause"
+	"github.com/sirupsen/logrus"
 )
 
-// Get all users last visited pages records
+// GetUsersLastVisitedPages returns all users last visited pages records
 func GetUsersLastVisitedPages(accountId uint) ([]models.LastVisitedPage, error) {
 	var pages []models.LastVisitedPage
 	err := database.DB.Order("updated_at desc").Where("user_identity_id = ?", accountId).Find(&pages).Error
 	return pages, err
 }
 
-// Check if the currently visited page is already in the visited pages array
-func CheckExistingPage(pages []models.LastVisitedPage, currentPage models.LastVisitedPage) bool {
-	pageExists := false
-	for _, page := range pages {
-		if page.Pathname == currentPage.Pathname {
-			pageExists = true
-			break
-		}
-	}
-	return pageExists
-}
-
-func HandlePostLastVisitedPages(accountId uint, currentPage models.LastVisitedPage) error {
-	// Try the entire thing in one transaction
-	return database.DB.Transaction(func(tx *gorm.DB) error {
-		var pages []models.LastVisitedPage
-		// Lock this row until updated
-		err := database.DB.Clauses(clause.Locking{Strength: "UPDATE"}).Order("updated_at desc").Where("user_identity_id = ?", accountId).Find(&pages).Error
-		if err != nil {
+// HandlePostLastVisitedPages inserts the most recent pages from chrome. Once they are added,
+// all older entries are removed from the table.
+func HandlePostLastVisitedPages(recentPages []models.LastVisitedPage, accountId uint) error {
+	var ids []uint
+	for _, v := range recentPages {
+		v.UserIdentityID = accountId
+		if err := database.DB.Create(&v).Error; err != nil {
```

Comment:
We should probably use the batch update gorm patter/: https://gorm.io/gen/create.html#Batch-Insert

It should generate just a single query and it should be a single transaction.